pkg/hubble/relay/observer/server.go

```diff
@@ -168,6 +169,61 @@ sortedFlowsLoop:
 	return g.Wait()
 }
 
+// GetNodes implements observerpb.ObserverClient.GetNodes.
+func (s *Server) GetNodes(ctx context.Context, req *observerpb.GetNodesRequest) (*observerpb.GetNodesResponse, error) {
+	if md, ok := metadata.FromIncomingContext(ctx); ok {
+		ctx = metadata.NewOutgoingContext(ctx, md)
+	}
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+	g, ctx := errgroup.WithContext(ctx)
+
+	peers := s.peers.List()
+	nodes := make([]*observerpb.Node, 0, len(peers))
+	for _, p := range s.peers.List() {
```

Comment:
Shouldn't we use `peers` here instead of calling `s.peers.List()` once again?

Reply:
Indeed, nice catch!